master/buildbot/www/authz/authz.py

```diff
@@ -89,10 +89,13 @@ def assertUserAllowed(self, ep, action, options, userDetails):
                     if owner is not None:
                         roles.update(
                             self.getOwnerRolesFromUser(userDetails, owner))
-                if rule.role not in roles:
-                    if rule.defaultDeny:
-                        raise Forbidden(
-                            "you need to have role '%s'" % (rule.role,))
-                elif not rule.defaultDeny:
-                    defer.returnValue(None)
+                for role in roles:
+                    if self.match(role, rule.role):
```

Comment:
I have question about tests.
In Authz class we have setUp method where self.authz.match (via authz.Authz constructor)
is set to fnmatchStrMatcher.
Is it ok to override self.authz.match to reStrMatcher in test method (after setUp)
 to also test regex matching?